functions/go/apply-replacements/main.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"sigs.k8s.io/kustomize/api/filters/replacement"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func main() {
+	if err := fn.AsMain(fn.ResourceListProcessorFunc(ApplyReplacements)); err != nil {
+		os.Exit(1)
+	}
+}
+
+func ApplyReplacements(rl *fn.ResourceList) error {
+	r := Replacements{}
+	return r.Process(rl)
+}
+
+type Replacements struct {
+	Replacements []types.Replacement `json:"replacements,omitempty" yaml:"replacements,omitempty"`
+}
+
+const fnConfigKind = "ApplyReplacements"
+const fnConfigApiVersion = "fn.kpt.dev/v1alpha1"
+
+// Config initializes Replacements from a functionConfig fn.KubeObject
+func (r *Replacements) Config(functionConfig *fn.KubeObject) error {
+	if functionConfig.Kind() != fnConfigKind || functionConfig.APIVersion() != fnConfigApiVersion {
+		return fmt.Errorf("received functionConfig of kind %s and apiVersion %s, " +
+			"only functionConfig of kind %s and apiVersion %s is supported",
+			functionConfig.Kind(), functionConfig.APIVersion(), fnConfigKind, fnConfigApiVersion)
+	}
+	r.Replacements = []types.Replacement{}
+	if err := functionConfig.As(r); err != nil {
+		return fmt.Errorf("unable to convert functionConfig to %s:\n%w",
+			"replacements", err)
+	}
+	return nil
+}
+
+// Process implements framework.ResourceListProcessor interface.
+func (r *Replacements) Process(rl *fn.ResourceList) error {
```

Comment:
@natasha41575 @mengqiy @yuwenma 

Regarding using this function as a library inside porch or kpt:

This is the interface https://github.com/GoogleContainerTools/kpt/blob/main/pkg/fn/eval.go#L27 porch or kpt CLI uses for injecting builtin function. It is basically at the wire-format level so that function-executor can be replaced by the builtin/remote/container-based executors.

So I see two levels:

1. Wireformat level (that takes in reader and writer streams) 
2. ResourceList data structure Level (the one defined by `Process` method here)

If we finalize the 2) as this `Processor` interface, then we can have an adaptor function in `kpt` or `porch` that implements 1) for each function. Here is an example (it uses framework....but we can migrate to use function SDK) --> https://github.com/GoogleContainerTools/kpt/blob/main/internal/builtins/pkg_context.go#L51

Reply:
I see. So to confirm, you are suggesting these functions implement `Process`, and kpt/porch can do the translation from Wireformat to ResourceList?

Reply:
Yes. Though, it will be good to have the helpers in fn-sdk.

Reply:
I can add some helpers to the sdk to do this. FYI @yuwenma

Reply:
+1 to add the helper in fn-sdk

Reply:
Filed https://github.com/GoogleContainerTools/kpt/issues/2897. Will start work on this shortly.